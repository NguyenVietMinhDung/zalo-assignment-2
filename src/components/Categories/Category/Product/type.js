import type { Product } from '../../type';

export type Props = {...Product};
