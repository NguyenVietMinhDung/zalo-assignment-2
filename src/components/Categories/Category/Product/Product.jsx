/* eslint-disable import/no-dynamic-require,global-require */
import React from 'react';
import './Product.scss';
import { Row, Col } from 'react-grid-system';
import Truncate from 'react-truncate';
import { Props } from './type';
import formatPrice from '../../../../utils/formatPrice';
import Ellipsis from '../../../Ellipsis';

const Product = (props: Props) => {
  const {
    name, price, img, description,
  } = props;
  const formattedPrice = formatPrice(price, 'de-DE', 'Ä‘');
  const imgSrc = img ? require(`../../../../assets/images/${img}`) : '';
  return (
    <Row className="product">
      <Col xs={3}>
        <img src={imgSrc} alt="" />
      </Col>
      <Col xs={6}>
        <div className="product__name">
          <Truncate lines={2} ellipsis={<Ellipsis />}>
            {name}
          </Truncate>
        </div>
        <div className="product__description">
          <Truncate lines={1} ellipsis={<Ellipsis />}>
            {description}
          </Truncate>
        </div>
      </Col>
      <Col xs={3}>
        {formattedPrice}
      </Col>
    </Row>
  );
};

export default Product;
