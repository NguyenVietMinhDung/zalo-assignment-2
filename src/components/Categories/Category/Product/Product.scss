@import '../../../../styles/variables';

.product {
	width: 100%;
	padding-top: 15px;
	padding-bottom: 15px;

	border-bottom: 1px solid $white-smoke;

	img {
		width: 100%;

		cursor: pointer;

		border-radius: 10px;
		background-color: $white-smoke;
	}

	&__name {
		margin-bottom: 5px;

		cursor: pointer;

		font-size: 14px;
		font-weight: 500;
	}

	&__description {
		opacity: 0.5;

		font-size: 10px;
	}
}
