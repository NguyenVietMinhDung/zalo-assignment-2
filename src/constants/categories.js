export default [
  {
    id: 'other',
    name: 'Khác',
    items: [
      {
        id: 0,
        name: 'Phin Inox',
        description: 'Phin Inox',
        price: 80000,
        img: '270_crop_PHININOX.png',
      },
      {
        id: 1,
        name: 'Ly Sứ Mosaic',
        description: 'Mosaic Mug',
        price: 29000,
        img: '270_crop_LYSU.png',
      },
    ],
  },
  {
    id: 'bread',
    name: 'Bánh Mì',
    items: [
      {
        id: 0,
        name: 'Xíu Mại',
        description: 'Pork Meatball',
        price: 19000,
        img: '270_crop_BMXIUMAI.png',
      },
      {
        id: 1,
        name: 'Chả Lụa Xá Xíu',
        description: 'Roast Pork & Pork Sausage',
        price: 19000,
        img: '270_crop_BMCHALUAXAXIU.png',
      },
      {
        id: 2,
        name: 'Gà Xé Nước Tương',
        description: 'Roast Chicken',
        price: 19000,
        img: '270_crop_BMGAXE.png',
      },
      {
        id: 3,
        name: 'Thịt Nướng',
        description: 'Grilled Pork',
        price: 19000,
        img: '270_crop_BMTHITNUONG.png',
      },
    ],
  },
  {
    id: 'cake',
    name: 'Bánh Ngọt',
    items: [
      {
        id: 0,
        name: 'Bánh Mousse Cacao',
        description: 'Cocoa Mousse Cake',
        price: 29000,
        img: '270_crop_MOUSSECACAO.png',
      },
      {
        id: 1,
        name: 'Bánh Mousse Đào',
        description: 'Peach Mousse Cake',
        price: 29000,
        img: '270_crop_MOUSSEDAO.png',
      },
      {
        id: 2,
        name: 'Bánh Sô-cô-la Highlands',
        description: 'Highlands Chocolate Bomb',
        price: 29000,
        img: '270_crop_SOCOLAHL.png',
      },
    ],
  },
  {
    id: 'coffee',
    name: 'Cà Phê Đóng Gói',
    items: [
      {
        id: 0,
        name: 'Latte',
        description: 'Latte',
        price: 60000,
        img: '270_crop_LATTE.png',
      },
      {
        id: 1,
        name: 'Cappuccino',
        description: 'Cappuccino',
        price: 60000,
        img: '270_crop_CAPPUCINO.png',
      },
      {
        id: 2,
        name: 'Americano',
        description: 'Americano',
        price: 60000,
        img: '270_crop_AMERICANO.png',
      },
      {
        id: 3,
        name: 'Espresso',
        description: 'Espresso',
        price: 60000,
        img: '270_crop_ESPRESSO.png',
      },
      {
        id: 4,
        name: 'Caramel Macchiato',
        description: 'Caramel Macchiato',
        price: 60000,
        img: '270_crop_CARAMEL-MACCHIATO.png',
      },
    ],
  },
  {
    id: 'Tea',
    name: 'Trà',
    items: [
      {
        id: 0,
        name: 'Trà Thạch Vãi',
        description: 'Lychee Tea',
        price: 39000,
        img: '270_crop_TRATHACHVAI_1.png',
      },
      {
        id: 1,
        name: 'Trà Thạch Đào',
        description: 'Peach Jelly Tea',
        price: 39000,
        img: '270_crop_TRATHACHDAO.png',
      },
      {
        id: 2,
        name: 'Trà Thanh Đào',
        description: 'Peach Lemongrass Tea',
        price: 39000,
        img: '270_crop_TRATHANHDAO.png',
      },
      {
        id: 3,
        name: 'Trà Sen Vàng',
        description: 'Golden Lotus Tea',
        price: 39000,
        img: '270_crop_TRASENVANG.png',
      },
    ],
  },
];
