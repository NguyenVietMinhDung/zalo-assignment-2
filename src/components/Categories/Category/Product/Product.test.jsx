import React from 'react';
import { shallow } from 'enzyme';
import Product from './Product';
import categories from '../../../../constants/categories';

const defaultProps = {
  ...categories[0][0],
};

describe('<Product />', () => {
  test('renders correctly', () => {
    const component = shallow(<Product {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
