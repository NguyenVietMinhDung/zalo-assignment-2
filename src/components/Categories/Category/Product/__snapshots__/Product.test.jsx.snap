// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> renders correctly 1`] = `
<Row
  align="normal"
  className="product"
  component="div"
  debug={false}
  gutterWidth={null}
  justify="start"
  nogutter={false}
  nowrap={false}
  style={Object {}}
>
  <Col
    component="div"
    debug={false}
    lg={null}
    md={null}
    offset={Object {}}
    pull={Object {}}
    push={Object {}}
    sm={null}
    style={Object {}}
    width={null}
    xl={null}
    xs={3}
    xxl={null}
  >
    <img
      alt=""
      src=""
    />
  </Col>
  <Col
    component="div"
    debug={false}
    lg={null}
    md={null}
    offset={Object {}}
    pull={Object {}}
    push={Object {}}
    sm={null}
    style={Object {}}
    width={null}
    xl={null}
    xs={6}
    xxl={null}
  >
    <div
      className="product__name"
    >
      <Truncate
        ellipsis={<_default />}
        lines={2}
        trimWhitespace={false}
        width={0}
      />
    </div>
    <div
      className="product__description"
    >
      <Truncate
        ellipsis={<_default />}
        lines={1}
        trimWhitespace={false}
        width={0}
      />
    </div>
  </Col>
  <Col
    component="div"
    debug={false}
    lg={null}
    md={null}
    offset={Object {}}
    pull={Object {}}
    push={Object {}}
    sm={null}
    style={Object {}}
    width={null}
    xl={null}
    xs={3}
    xxl={null}
  >
    NaNÄ‘
  </Col>
</Row>
`;
